### PR TITLE
coprocessor scalar function is_ipv4 implementation

### PR DESCRIPTION
**Pull request by opensourcegeek**
## What have you changed? (mandatory)

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:
- Summarize your change (mandatory)
is_ipv4 scalar function is implemented 

- How does this PR work? Need a brief introduction for the changed logic (optional)
New builtin_miscellaneous file is added to coprocessor/dag/expr/ directory and added another `impl ScalarFunc` to add is_ipv4 method to it. This evaluates the incoming string and checks whether it is a IPv4 address or not and the relevant tests are added.

- Separately describe each logical change and avoid lazy messages (optional)

- Describe any limitations of the current code (optional)

Please **NOTE** that:
- Do not assume reviewers understand the original issue

## What are the type of the changes? (mandatory)

- New feature (non-breaking change which adds functionality)

## How has this PR been tested? (mandatory)
Unit tests are added within the module

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

## Refer to a related PR or issue link (optional)
https://github.com/tikv/tikv/issues/3275

### CODE REVIEW

**Comment by siddontang** [comment]
Thanks @opensourcegeek 

Please sign off your all commits. See https://github.com/tikv/tikv/blob/master/CONTRIBUTING.md#signing-off-the-commit

If you have already pushed some commits before, you can use `git rebase -i master --signoff` to resign them off.

**Comment by opensourcegeek** [comment]
@siddontang I've edited the commit message and pushed to my branch

File: src/coprocessor/dag/expr/scalar_function.rs [hunk 1]
```diff
@@ -190,7 +190,9 @@ impl ScalarFunc {
             | ScalarFuncSig::CRC32
             | ScalarFuncSig::JsonTypeSig
             | ScalarFuncSig::JsonUnquoteSig
+            | ScalarFuncSig::IsIPv4
             | ScalarFuncSig::ASCII
+            | ScalarFuncSig::IsIPv4
```
Comment (AndreMouche) [inline src/coprocessor/dag/expr/scalar_function.rs]: Why two `IsIpv4`?

File: src/coprocessor/dag/expr/builtin_miscellaneous.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+use std::net::IpAddr;
+use std::str::FromStr;
+
+use super::{EvalContext, Result, ScalarFunc};
+use coprocessor::codec::Datum;
+
+impl ScalarFunc {
+    pub fn is_ipv4(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let input = try_opt!(self.children[0].eval_string(ctx, row));
+        let s = String::from_utf8(input.into_owned())?;
```
Comment (AndreMouche) [inline src/coprocessor/dag/expr/builtin_miscellaneous.rs]: Could we use `eval_string_and_decode` instead?

File: src/coprocessor/dag/expr/builtin_miscellaneous.rs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+#[cfg(test)]
+mod test {
+    use super::is_given_string_ipv4;
+    use coprocessor::codec::Datum;
+    use coprocessor::dag::expr::test::{datum_expr, scalar_func_expr};
+    use coprocessor::dag::expr::{EvalContext, Expression};
+    use tipb::expression::ScalarFuncSig;
+
+    #[test]
+    fn test_is_ipv4_success() {
```
Comment (AndreMouche) [inline src/coprocessor/dag/expr/builtin_miscellaneous.rs]: Could we merge `test_is_ipv4***` into one case?

**Comment by opensourcegeek** [comment]
The author names are incorrect and this is cause the DCO checks to fail. I'll fix these correctly and resubmit a fresh PR. Please delete this one.

**Pull request closed without merging**